### PR TITLE
Avoid sync require for loading NLS configuration (fix #159894)

### LINKED ISSUE #159894: Avoid sync `require` for loading NLS configuration
Will provide a PR for this shortly.

### PR DESCRIPTION
**Pull request by bpasero**
It would be great if I can learn how to validate this works running out of sources.

Hope the code change is fine, I think a `require` in node.js is loading the file from disk and `JSON.parse()` it if its a JSON file.

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
I'll approve and we can try it out in insiders next week but let me see what the local testing options are

File: src/vs/base/node/languagePacks.js [hunk 1]
```diff
@@ -141,94 +141,95 @@
...
-					return defaultResult(initialLocale);
-				}
-				const packConfig = configs[locale];
-				let mainPack;
-				if (!packConfig || typeof packConfig.hash !== 'string' || !packConfig.translations || typeof (mainPack = packConfig.translations['vscode']) !== 'string') {
-					return defaultResult(initialLocale);
-				}
-				return exists(mainPack).then(fileExists => {
-					if (!fileExists) {
+				return getLanguagePackConfigurations(userDataPath).then(configs => {
```
Comment (TylerLeonhardt) [inline src/vs/base/node/languagePacks.js:144]: about time we moved to async/await, huh :)

**Review by TylerLeonhardt — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
Tried testing this but it's pretty tough... let's see what Insiders ends up looking like

**Comment by dbaeumer** [comment]
The only way to test this is to build a VS Code version locally. The code is skipped when running out of source.

**Pull request merged**